### DISCUSSION TITLE
Limit display length of solutions

### DISCUSSION THREAD

**Discussion by ghost** [codewars/codewars.com#2827] [Ideas]
As amusing as it is to write lengthy and obscure solutions when we get board, one shouldn't have to read (e.g.) 50 pages of magic numbers to see the solutions below it.  Case in point:

https://www.codewars.com/kata/reviews/57626fb4eb1e55a4120023f2/groups/60221ffb6aef7e0001187389

**Comment by kazk** [comment]
Done

**Comment by Madjosz** [comment]
While the current implementation is an improvement dicussions have been raised on Discord about the general way how this feature could be further improved. I want to note some of the points here:
* The current implementation puts a solution into a scrollable container with 500px max height. This seems a little bit too small as it e.g. only fits 23 lines of code in my browser which is not very much and is normally already reached by a standard 4kyu solution. Especially since most users requested this feature to get rid of annoying 1000-line-hardcoded-magic-numbers-troll-solutions, so the limits could be a little bit higher.
* The fact that we now have a scrollable component in a scrollable component can make it hard for mobile users because they still have to scroll the whole solution when the inner component fills the whole viewport.
* Also the visibility that "there is more to that solution" is somewhat harder because you either have to recognize that there are cut-off lines at the bottom or recognize the scroll bar where you have to be lucky that your browser even renders a noticable scrollbar when you aren't focussing/hovering the scroll-pane or even the 'scrollbar-area'.
* A preferred solution for me would be to replace the vertical scroll functionality completely by a "show more" functionality which expands the complete solution on button click like e.g. on Quora:
![Quora "Continue Reading" functionality](https://user-images.githubusercontent.com/28844868/202465416-b36aeb16-9296-4a03-8328-8843c63a4960.png)

**Comment by JohanWiltink** [comment]
If it's at all possible, could the box height depend on the screen size? _Just_ under a screenfull of code would be a nice size.

**Comment by kazk** [comment]
I like @Madjosz's "show more", but I'll make the max height 95% of the viewport height for now.
